stdlib/Sockets/src/addrinfo.jl

```diff
@@ -251,3 +251,39 @@ function getipaddr()
     ccall(:uv_free_interface_addresses, Cvoid, (Ptr{UInt8}, Int32), addr, count)
     return lo_present ? localhost : error("No networking interface available")
 end
+
+"""
+    getipaddrs() -> Vector{IPv4}
+
+Get the IP addresses of the local machine.
+
+# Examples
+```julia-repl
+julia> getipaddrs()
+2-element Array{IPv4,1}:
+ ip"10.255.0.183"
+ ip"172.17.0.1"
+```
+"""
+function getipaddrs()
+    addresses = IPv4[]
+    addr_ref = Ref{Ptr{UInt8}}(C_NULL)
+    count_ref = Ref{Int32}(1)
+    lo_present = false
+    err = ccall(:jl_uv_interface_addresses, Int32, (Ref{Ptr{UInt8}}, Ref{Int32}), addr_ref, count_ref)
+    uv_error("getlocalip", err)
+    addr, count = addr_ref[], count_ref[]
+    for i = 0:(count-1)
+        current_addr = addr + i*_sizeof_uv_interface_address
+        if 1 == ccall(:jl_uv_interface_address_is_internal, Int32, (Ptr{UInt8},), current_addr)
+            lo_present = true
```

Comment:
dropping these seems odd to me. seems like these should probably be returned (with each entry being a namedtuple), or a perhaps configurable by function argument

Reply:
The current behavior seems like it is a reasonable default, so we can add an option to list internal interfaces as well as external ones at a future time, although it would not be a bad addition to make in this PR given that this new feature won't land in a release until 1.2 in the spring.

Reply:
Does my latest changes resolve this issue?